tests/localosmosis/Dockerfile

```diff
@@ -0,0 +1,91 @@
+# syntax=docker/dockerfile:1
+
+ARG BASE_IMG_TAG=nonroot
+
+# --------------------------------------------------------
+# Build
+# --------------------------------------------------------
+
+FROM golang:1.18.2-alpine3.15 as build
+
+RUN set -eux; apk add --no-cache ca-certificates build-base;
+RUN apk add git
+# Needed by github.com/zondax/hid
+RUN apk add linux-headers
+
+WORKDIR /osmosis
+COPY . /osmosis
+
+# CosmWasm: see https://github.com/CosmWasm/wasmvm/releases
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
+RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 7d2239e9f25e96d0d4daba982ce92367aacf0cbd95d2facb8442268f2b1cc1fc
+RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep f6282df732a13dec836cda1f399dd874b1e3163504dbd9607c6af915b2740479
+
+# CosmWasm: copy the right library according to architecture. The final location will be found by the linker flag `-lwasmvm_muslc`
+RUN cp /lib/libwasmvm_muslc.$(uname -m).a /lib/libwasmvm_muslc.a
+
+RUN BUILD_TAGS=muslc LINK_STATICALLY=true make build
+
+# --------------------------------------------------------
+# Runner
+# --------------------------------------------------------
+
+FROM ubuntu
+
+COPY --from=build /osmosis/build/osmosisd /bin/osmosisd
+
+ENV HOME /osmosis
+WORKDIR $HOME
+RUN apt-get update && apt-get install -y jq
+# change staking denom to uosmo
+RUN osmosisd init --chain-id=localosmosis val
+RUN echo "satisfy adjust timber high purchase tuition stool faith fine install that you unaware feed domain license impose boss human eager hat rent enjoy dawn" | osmosisd keys add val --recover --keyring-backend=test
+RUN cat $HOME/.osmosisd/config/genesis.json | jq '.app_state["staking"]["params"]["bond_denom"]="uosmo"' > $HOME/.osmosisd/config/tmp_genesis.json && mv $HOME/.osmosisd/config/tmp_genesis.json $HOME/.osmosisd/config/genesis.json
+RUN osmosisd add-genesis-account osmo1phaxpevm5wecex2jyaqty2a4v02qj7qmlmzk5a 100000000000uosmo,100000000000uion
```

Comment:
Each RUN in a Dockerfile adds an additional layer, making the images large in size:
https://stackoverflow.com/questions/39223249/multiple-run-vs-single-chained-run-in-dockerfile-which-is-better

We should extract all these in a bash script and execute that instead

Reply:
Ah, see I didn't know that. I had these in a script file to begin with but then felt like I might as well combine them in the required dockerfile as well. I will separate and instead call that bash file from the dockerfile, thanks!